javascript/javascript1/week4/homework6/scripts.js

```diff
@@ -0,0 +1,133 @@
+    return console.log(`Today is the ${today.getDate()}. of ${months[month]} ${today.getFullYear()}`)
+}
+
+// function for timer 
+function timer(command) {
+    function getInterval(command){
+        const regex = /(\d+)/ //regular expression to find digits in the command str
+        const match = command.match(regex) // this method will return an array with matching values to the regex 
+        return match ? Number(match[0]) : null; // if true convert the regex to a number and return it
+    }
+    const number = getInterval(command)
+    const interval = number * 60 * 1000;
+    function logDone(){
+        console.log(`Timer Done`)
+    }
+    setTimeout(logDone, interval)
+    console.log(`A timer for ${number} minute(s) has been set`)
+}
+
+
+
+//main function to get answers from the "voice assistant"
+function getReply(command) {
+    // here we deal with the greeting commands
+    if (command.startsWith('Hello my name is')){
+        const person = getName(command)
+        if (!loggedName.includes(person)) {
+            loggedName.push(person)
+            return console.log(`Nice to meet you ${person}`)
+        } else if (loggedName.includes(person)) {
+            return console.log(`Hey ${person}! we met already.`)
+        }
+    }
+    // here we deal with reminding the user their name
+    else if (command.startsWith('What is my name')) {
+        if (loggedName.length === 0) {
+            return console.log(`I can't find a name. \n Please introduce yourself in this way: "Hello my name is" followed by your name`)
+        } else if (loggedName.length > 0) {
+            const recentName = loggedName[loggedName.length -1]
+            return console.log(`Your name is ${recentName}! What can I help you with, ${recentName}?`)
+        }
+    }
+    //dealing with the todo part
+    else if (command.includes('my todo')) {
+        let newTask = getTask(command);
+        //adding a task
+        if(command.startsWith('Add') || command.startsWith('add')){
+           todos.push(newTask); 
+           return console.log(`${newTask} has been added to your todos`)
+        
+        } 
+        //removing a task
+        else if (command.startsWith('Remove') || command.startsWith('remove')){
+            const taskIndex = todos.indexOf(newTask);
+            todos.splice(taskIndex, 1)
+            return console.log(`${newTask} has been removed from your todos`)
+        }
+        // showing all task 
+        else if (command.startsWith('What is on my todo?')){
+            if (todos.length === 0) {
+                console.log(`Your list is empty, try adding some tasks first`)
+            } else {
+                console.log(`Your todo list is comprised of: \n${todos.join('\n')}`)
+            }
+        }
+    }
+    // dealing with the date 
+    else if (command.startsWith('What day is it today?') || command.startsWith('What day is it') ) {
```

Comment:
The second part of this statement will always be true if the first half is, so we may wish to drop the first part entirely!

Reply:
Another oversight. Thank you for pointing out the redundancy🙏